File: codeWars/katas/binary_divisible_by/binary_divisible_by.js
// Binary Divisible By ?


function binaryDivisibleBy(str, divisible = 3) {
    // regex for only 0 and 1 number
    const regex = /^[0-1]+$/

    // check str is 0 and 1 only
    const isZeroOne = str.match(regex);

    if (isZeroOne) {
        // converting into number
        const convertToNum = parseInt(isZeroOne, 2);

        // return true if number is divisible by 3
        if (convertToNum % divisible === 0) return true;
        else return false;
    } else {
        return false;
    }
}

const textCase1 = binaryDivisibleBy("000")
console.log(textCase1)

const textCase2 = binaryDivisibleBy("011")
console.log(textCase2)

const textCase3 = binaryDivisibleBy("001")
console.log(textCase3)

File: codeWars/katas/binary_divisible_by/binary_divisible_by.py
import re


def binaryDivisibleBy(string, divisible=3):
    # Regular expression for only 0 and 1 digits
    regex = r"^[01]+$"

    # Check if string contains only 0s and 1s
    is_zero_one = re.match(regex, string)

    if is_zero_one:
        # Convert string to integer
        convert_to_num = int(is_zero_one.group(), 2)

        # Check if number is divisible by divisible
        if convert_to_num % divisible == 0:
            return True
        else:
            return False
    else:
        return False


textCase1 = binaryDivisibleBy("000")
print(textCase1)

textCase2 = binaryDivisibleBy("011")
print(textCase2)

textCase3 = binaryDivisibleBy("001")
print(textCase3)
